### ISSUE TITLE
Antivirus detects possible malware "IDP.Generic" in Windows version (x64)

### ISSUE THREAD

**Issue by RafaelLinux** [zaps166/QMPlay2#636]
It's a headache to install QMPlay2 on Windows.

1. Browser warns about possible risk security
2. Windows warns about insecure application
3. Despite user ignore those warnings and install it, finally Avast (surely some other antivirus) warn about to block QMplay2 to run cause it has "IDP.Generic" malware. 
4. User blocks QMPlay2 and uninstall

Did you know this issue? 

Thanks

**Comment by zaps166** [comment]
- I don't use antiviruses on Windows (including Defender), they are slowing down PC and shows too much false-positives
- I don't pay for certificates to use them in QMPlay2 on Windows, so antiviruses might show everything
- QMPlay2 is cross-compilled from Arch Linux mingw-w64 package, I hope it's clean and doesn't contain any viruses
- you might have infected installer - 64-bit Windows installer has `075f1dd6aaeba5307a2c35f4c72085548a7094e7ab7936895752e7840b193d4f` checksum
- you can use https://www.virustotal.com/gui/home/upload and upload QMPlay2 exe and DLL files and check there

As you can see, QMPlay2.exe is clean: https://www.virustotal.com/gui/file/095f0b1c5d5e8ee928d1787e76c8983439ea52601eae6298a6a466802ed57b5b
Installer is also clean (except one false-positive): https://www.virustotal.com/gui/file/075f1dd6aaeba5307a2c35f4c72085548a7094e7ab7936895752e7840b193d4f
Main DLL is also clean https://www.virustotal.com/gui/file/0c800da47545865f064f14ef7570825f0bd2a3439959e41e6c38d6eee23acd1f

You can see all files if you want.

> Browser warns about possible risk security

Which browser? Firefox or Edge?

> Windows warns about insecure application

I don't use certificates, so Windows will warn, it's expected. I don't want to pay for any certificates. If there are also free certs, I can use it, but somebody have to guide me.

---

I've never had issues on clean Windows installation (with Windows defenter, too), so it must be your configuration (Avast?).

**Comment by RafaelLinux** [comment]
>   * I don't use antiviruses on Windows (including Defender), they are slowing down PC and shows too much false-positives
Really, I'm not a Windows user, but I understand to have antivirus software it's essential for users that are connected to Internet. Nowadays more than ever it is very easy to trick Windows users into infecting their computers, I notice it every time I use Windows and see the amount of misleading windows that appear pretending to be system messages and trick the user into falling into the trap. 
> 
>     * I don't pay for certificates to use them in QMPlay2 on Windows, so antiviruses might show everything
I fully understand that, as any free and open source developer, you don't pay a company that does charge for its licenses and that, therefore, QMPlay2 is not in the MS "store".

>     * QMPlay2 is cross-compilled from Arch Linux mingw-w64 package, I hope it's clean and doesn't contain any viruses
Perhaps it is that compiler that generates the code that is identified as malicious.

>     * you might have infected installer - 64-bit Windows installer has `075f1dd6aaeba5307a2c35f4c72085548a7094e7ab7936895752e7840b193d4f` checksum
I dowloaded it directle from GItHub.

>     * you can use https://www.virustotal.com/gui/home/upload and upload QMPlay2 exe and DLL files and check there
> As you can see, QMPlay2.exe is clean: https://www.virustotal.com/gui/file/095f0b1c5d5e8ee928d1787e76c8983439ea52601eae6298a6a466802ed57b5b Installer is also clean (except one false-positive): https://www.virustotal.com/gui/file/075f1dd6aaeba5307a2c35f4c72085548a7094e7ab7936895752e7840b193d4f Main DLL is also clean https://www.virustotal.com/gui/file/0c800da47545865f064f14ef7570825f0bd2a3439959e41e6c38d6eee23acd1f
> You can see all files if you want.

Thank you very much, I know that page. This thread is only intended to let you know that, for a Windows user (to whom I recommend QMPlay2) all those obstacles I describe, and especially the virus detection, make the user (in my opinion, with good judgement not to take risks) desist from installing a good application such as yours. In fact, the "IDP.Generic" malware includes any software that can steal password information, among other things.  Having gone through my own and others' experience with the problems of malware intrusion into Windows, if someone asks me what to do, I would clearly say "Don't install it". 

> 
> > Browser warns about possible risk security
> Which browser? Firefox or Edge?

I'm sorry, I didn't tell you: Firefox, Windows version.

> > Windows warns about insecure application
> 
> I don't use certificates, so Windows will warn, it's expected. I don't want to pay for any certificates. If there are also free certs, I can use it, but somebody have to guide me.

Sincerely, I have no idea about that, and do not pretend you spend any money on that.

> I've never had issues on clean Windows installation (with Windows defenter, too), so it must be your configuration (Avast?).

Yes, I usually recommend Avast and this PC have it installed.

Anyway, my intention is only inform you about this issue in Windows.

**Comment by zaps166** [comment]
I'll be able to check on clean Windows soon.

**Comment by zaps166** [comment]
I was able to download and install on Windows 11 - only one Windows security screen which is expected for non-certified downloaded software. Firefox didn't warn, Microsoft Defender also didn't warn.

**Comment by zaps166** [comment]
Avast is not showing anything on VirusTotal, maybe you hev custom settings like heuristic?

Looks like "IDP.Generic" is just false-positive, Avast doesn't know what's this.

**Comment by RafaelLinux** [comment]
I needed time to try it again in a Windows system. 
Using EDGE to dowload, first warning:
![image](https://github.com/zaps166/QMPlay2/assets/13048959/b4709cd0-5c72-476b-b984-d7fdc02d33b7)
Translation is something like "This is not a common download. Be sure you rely on ..." so to execute, user must first follow arrows sequence marked in screenshot. After click in "Keep" ("Conservar") a new window:
![image](https://github.com/zaps166/QMPlay2/assets/13048959/eb631dc1-e1f3-43ac-b0fc-02d1a01062ca)
that only shows two big buttons, "Remove" or "Cancel". An average user even don't imagine he must click in "Show more" ("Mostrar más") to continue the adventure to get to install this external application. After click in "Keep anyway" ("Conservar de todos modos") that is one of the new options showed, user must click again in "Open file" (good news!!)
![image](https://github.com/zaps166/QMPlay2/assets/13048959/10ac37b5-01cd-4bc4-9eb3-079e595cdba4)

... but , new trouble. A giant Windows window warning about "Windows protected your PC!!" and only a great button ... "NO execute". At this point, user can be bored or scared or lost or all them. "How in hell then can I install it?" can user ask herself.
If a dare and smart user, he will click "More info" and the a new button "Execute anyway" appears. Eureka. Finally, after some other typicall questions about installation, I got to install QMPlay2 and (in this computer, with Avast antivirus too) Avast doesn't warn anything about QMPlay2 when installing or executing it.

So after my relate, maybe you could include a link or reference for Windows users that pretend to install QMPlay2.

;)

**Comment by zaps166** [comment]
> Avast doesn't warn anything about QMPlay2 when installing or executing it.

It shouldn't, now it's correct! :slightly_frowning_face: 

---

So with Edge we have 2 security warnings, with Firefox 1 security warning :smile: 

---

Btw. where did you found QMPlay2?

**Comment by RafaelLinux** [comment]
> > Avast doesn't warn anything about QMPlay2 when installing or executing it.
> 
> It shouldn't, now it's correct! slightly_frowning_face
> 
> So with Edge we have 2 security warnings, with Firefox 1 security warning smile

Plus Windows (SmartScreen) warning

> Btw. where did you found QMPlay2?

I didn't remember, but maybe thru Linux webpages talking about Linux applications. I used VLC, but it's interface never liked to me. Then I switched to SMPlayer but no easy to install Windows version and an issue that always happened and was not solved make me search for another player and 3 years ago I read about QMPlay2, tried it and it's just was I was looking for plus incredible features.
